xml_converter/tests/attribute_testing.py

```diff
@@ -0,0 +1,140 @@
+from dataclasses import dataclass
+import subprocess
+import re
+import os
+from typing import Dict, List
+
+# Path to compiled C++ executable
+xml_converter: str = "../build/xml_converter"
+
+
+@dataclass
+class Attribute_Tests:
+    attribute_name: str
+    attribute_path: str
+    tests: List[str]
+    output_result: Dict[str, List[str]]
+    expect_output_to_be_equal: Dict[str, bool]
+    expect_files_to_be_equal: Dict[str, bool]
+
+
+def run_xml_converter(*args):
+    try:
+        # Build the command to execute the C++ program with the desired function and arguments
+        cmd = [xml_converter] + list(args)
+
+        # Run the C++ program and capture its output
+        result = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+
+        # Check if the C++ program executed successfully
+        if result.returncode == 0:
+            return result.stdout
+        else:
+            return result.stderr
+    except Exception as e:
+        print("Error:", e)
+        return None
+
+
+def are_files_equal(file_path1, file_path2) -> bool:
+    try:
+        with open(file_path1, 'rb') as file1, open(file_path2, 'rb') as file2:
+            content1 = file1.read()
+            content2 = file2.read()
+
+        return content1 == content2
+
+    except (FileNotFoundError, PermissionError):
+        print(f"Error opening file {file_path1} or {file_path2}")
+        return False
+
+
+def are_arrays_equal(array1, array2) -> bool:
+    if len(array1) != len(array2):
+        return False
+
+    for i in range(len(array1)):
+        if array1[i] != array2[i]:
+            return False
+
+    return True
+
+
+arg_input_xml: str = "--input-taco-path"
+arg_output_xml: str = "--output-taco-path"
+
+chrono_patterns = [
+    "^The taco parse function took [0-9]+ milliseconds to run$",
+    "^The xml write function took [0-9]+ milliseconds to run$",
+    "^The protobuf write function took [0-9]+ milliseconds to run$"
+]
+
+
+if __name__ == "__main__":
+    # TODO: This will need to be extracted as more attributes and tests are added
+    attribute: Attribute_Tests = Attribute_Tests(
+        attribute_name="can_fade",
+        attribute_path="../test_cases/can_fade_tests",
+        tests=["_is_correct", "_is_different_file", "_is_wrong_output"],
+        output_result={
+            "_is_correct": ['Loading taco pack ../test_cases/can_fade_tests/can_fade_is_correct.xml'],
+            "_is_different_file": ['Loading taco pack ../test_cases/can_fade_tests/can_fade_is_different_file.xml'],
+            "_is_wrong_output": ['WRONG']
+        },
+        expect_output_to_be_equal={
+            "_is_correct": True,
+            "_is_different_file": True,
+            "_is_wrong_output": False,
+        },
+        expect_files_to_be_equal={
+            "_is_correct": True,
+            "_is_different_file": False,
+            "_is_wrong_output": True,
+        }
```

Comment:
can this be something like
```python
attribute: Attribute_Tests = Attribute_Tests(
...
    tests=[
       {output_results="", expect_output_to_be_equal="", expect_files_to_be_equal=""},
       ...
    ]
...
)
```
maybe even allow them to be more then just true/false values
* `expected_output_file_reference: Optional[str]`
* `expected_stdout: Optional[str]`
* `expected_stderr: Optional[str]`

Reply:
I realized that since we are going to be explicitly stating what we expect the outputs to be, that things like ```expect_output_to_be_equal = false``` are not testing the attribute themselves but the test program. I don't think that is necessary for the goal of this script. 

I changed the data format to be as follows. 

```
tests": {
            "_is_correct": {
                "output_stdout": [
                    "Loading taco pack ../test_cases/can_fade_tests/can_fade_is_correct.xml"
                ],
                "output_stderr": [],
                "expected_xml_diff": []
            },
```